src/System.Text.Json/tests/Utf8JsonWriterTests.cs

```diff
@@ -862,37 +862,51 @@ public void WritingTooDeepProperty(bool formatted, bool skipValidation)
         [InlineData(false, false)]
         public void WritingTooLargeProperty(bool formatted, bool skipValidation)
         {
+            if (IntPtr.Size < 8)
+            {
+                // Do not run these tests on X86 architecture since they will almost certainly OOM.
+                return;
+            }
+
             var state = new JsonWriterState(options: new JsonWriterOptions { Indented = formatted, SkipValidation = skipValidation });
 
             var output = new ArrayBufferWriter(1024);
 
             var jsonUtf8 = new Utf8JsonWriter(output, state);
 
-            Span<byte> key = new byte[1_000_000_000];
-            key.Fill((byte)'a');
-
-            var keyChars = new char[1_000_000_000];
-            keyChars.AsSpan().Fill('a');
-
             try
             {
-                jsonUtf8.WriteStartObject();
-                jsonUtf8.WriteStartArray(keyChars);
-                Assert.True(false, "Expected ArgumentException to be thrown for depth >= 1000.");
-            }
-            catch (ArgumentException) { }
 
-            jsonUtf8 = new Utf8JsonWriter(output, state);
+                Span<byte> key = new byte[1_000_000_000];
+                key.Fill((byte)'a');
 
-            try
+                var keyChars = new char[1_000_000_000];
+                keyChars.AsSpan().Fill('a');
+
+                try
+                {
+                    jsonUtf8.WriteStartObject();
+                    jsonUtf8.WriteStartArray(keyChars);
+                    Assert.True(false, "Expected ArgumentException to be thrown for depth >= 1000.");
+                }
+                catch (ArgumentException) { }
+
+                jsonUtf8 = new Utf8JsonWriter(output, state);
+
+                try
+                {
+                    jsonUtf8.WriteStartObject();
+                    jsonUtf8.WriteStartArray(key);
+                    Assert.True(false, "Expected ArgumentException to be thrown for depth >= 1000.");
+                }
+                catch (ArgumentException) { }
+
+                output.Dispose();
+            }
+            catch (OutOfMemoryException)
             {
-                jsonUtf8.WriteStartObject();
-                jsonUtf8.WriteStartArray(key);
-                Assert.True(false, "Expected ArgumentException to be thrown for depth >= 1000.");
+                return;
             }
```

Comment:
What do we expect to OOM?  Just the new byte/char[] allocations, or something in the json writer as well?  If the former, can we move the catch block up so that it's not hiding any OOM issues that might occur in the latter?